app/scripts/controllers/permissions/loggerMiddleware.js

```diff
@@ -81,51 +81,58 @@ module.exports = function createLoggerMiddleware ({
     return Object.keys(request.params[0])
   }
 
-  function logHistory (requestedMethods, origin, result, time) {
-
-    let accounts
-    const entries = result
-      ? result.map(perm => {
-        if (perm.parentCapability === 'eth_accounts') {
-          accounts = getAccountsFromPermission(perm)
-        }
-        return perm.parentCapability
-      })
-        .reduce((acc, m) => {
-          if (requestedMethods.includes(m)) {
-            acc[m] = {
-              lastApproved: time,
-            }
-            if (m === 'eth_accounts') {
-              acc[m].accounts = accounts
-            }
+  function logHistory (requestedMethods, origin, result, time, isEthRequestAccounts) {
+    let accounts, entries
+    if (isEthRequestAccounts) {
+      accounts = result
+      const accountToTimeMap = accounts.reduce((acc, account) => ({ ...acc, [account]: time }), {})
+      entries = { 'eth_accounts': { accounts: accountToTimeMap, lastApproved: time } }
+    } else {
+      entries = result
+        ? result.map(perm => {
+          if (perm.parentCapability === 'eth_accounts') {
+            accounts = getAccountsFromPermission(perm)
           }
-          return acc
-        }, {})
-      : {}
+          return perm.parentCapability
+        })
+          .reduce((acc, m) => {
+            if (requestedMethods.includes(m)) {
+              if (m === 'eth_accounts') {
+                const accountToTimeMap = accounts.reduce((acc, account) => ({ ...acc, [account]: time }), {})
```

Comment:
Nit: This is the only place `accounts` is used. Instead of setting that variable in the earlier `.map` step, it'd be simpler to get it here. Then the function-scoped variable wouldn't be needed anymore either.

Reply:
Not sure I follow. The `.map` returns an array of strings that are `permission.parentCapability`s, whereas to get `accounts` we need to pass a `permission` object to `getAccountsFromPermission`. How are you thinking we could get accounts within the `.reduce`?

Reply:
Ah I see, sorry I missed that the _entire_ `permission` object was needed for `getAccountsFromPermission`. Disregard this suggestion.